packages/server/src/unstable-core-do-not-import/stream/utils/asyncIterable.ts

```diff
@@ -27,6 +27,7 @@ export async function* withCancel<T>(
 interface TakeWithGraceOptions {
   count: number;
   gracePeriodMs: number;
+  onCancel?: () => void;
```

Comment:
Can't this be an abort signal instead of a callback?

Also, I think it's only used in one place, no need to make it optional

Reply:
It could be an abort ctrl passed in. But I like to not make util functions too specific. If there's a simpler or more generic way, without adding pretty much any effort, it's best to keep util functions generic or simple. In this case, I'd argue it's both more simple and generic. Lean function headers are great.